lib/python/qmk/cli/compile.py

```diff
@@ -28,6 +29,24 @@ def compile(cli):
     If --keyboard and --keymap are provided this command will build a firmware based on that.
 
     """
+    # Set CWD as directory command was issued from
+    cwd = os.path.normpath(os.path.join(os.environ['ORIG_CWD']))
```

Comment:
You should just use `os.environ['ORIG_CWD']` directly. The path was normalized at the time it was stored.